htlcswitch/hop/iterator.go

```diff
@@ -294,12 +295,24 @@ func (p *OnionProcessor) DecodeHopIterators(id []byte,
 		return lnwire.CodeNone
 	}
 
-	for i, req := range reqs {
-		onionPkt := &onionPkts[i]
-		resp := &resps[i]
+	// Execute cpu-heavy onion decoding in parallel.
+	var wg sync.WaitGroup
+	for i := range reqs {
+		// Copy loop variable for safe binding to goroutine.
+		seqNum := uint16(i)
```

Comment:
nit: could pass in i as an argument to the goroutine instead.

Reply:
Nice, fixed.